File: PolicyPlus/EditPolStringData.vb
Public Class EditPolStringData
    Public Function PresentDialog(ValueName As String, InitialData As String) As DialogResult
        TextName.Text = ValueName
        TextData.Text = InitialData
        TextData.Select()
        TextData.SelectAll()
        Return ShowDialog()
    End Function
    Private Sub EditPolStringData_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.Escape Then DialogResult = DialogResult.Cancel
    End Sub
End Class

File: PolicyPlus/EditPolStringData.cs
using System.Windows.Forms;

namespace PolicyPlus
{
    public partial class EditPolStringData
    {
        public EditPolStringData()
        {
            InitializeComponent();
        }
        public DialogResult PresentDialog(string ValueName, string InitialData)
        {
            TextName.Text = ValueName;
            TextData.Text = InitialData;
            TextData.Select();
            TextData.SelectAll();
            return ShowDialog();
        }
        private void EditPolStringData_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
                DialogResult = DialogResult.Cancel;
        }
    }
}